File: backend/Galmuri/src/main/java/com/ssafy/galmuri/domain/user/UserTrip.java
package com.ssafy.galmuri.domain.user;

import com.ssafy.galmuri.domain.trip.Trip;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Getter
@NoArgsConstructor
@IdClass(UserTripID.class)
@Entity
public class UserTrip {
    @Id
    @ManyToOne
    @JoinColumns({
            @JoinColumn(name = "userId"),
            @JoinColumn(name = "domain")
    })
    private User user;

    @Id
    @ManyToOne
    @JoinColumn(name = "tripId" )
    private Trip trip;

    @Column(nullable = false)
    private boolean isConfirmed;

    @Column(nullable = false)
    private boolean isOwn;
}

File: mobile/app/src/main/java/com/ssafy/gumi107/mobile_app/dto/UserTrip.kt
package com.ssafy.gumi107.mobile_app.dto

data class UserTrip( // 변수이름 카멜로 해라!
    var isConfirmed : Boolean, // 여행의 참가자인가?
    var isOwn : Boolean, // 여행 주최자(호스트) 인가?
    var userId : String, // PK FK 오직 1글자
    var domain : String, // PK FK 최대 255글자
    var tripId : Long, // PK FK
)
